fireant/slicer/transformers/datatables.py

```diff
@@ -274,16 +270,29 @@ def _render_column_level(self, metric_column, display_schema):
         }
 
     def _recurse_dimensions(self, dataframe, dimensions, metrics, reference=None):
+        if dataframe.empty:
+            return []
+
         if reference or not dimensions:
             return super(DataTablesColumnIndexTransformer, self)._recurse_dimensions(dataframe, dimensions, metrics,
                                                                                      reference)
 
+        data = []
+
         if 'display_field' in dimensions[0][1]:
             return [(key, dict(self._recurse_dimensions(dataframe[:, key, display], dimensions[1:], metrics)))
                     for key, display in zip(*dataframe.index.levels[1:3])]
 
-        return [(str(level), dict(self._recurse_dimensions(dataframe[:, level], dimensions[1:], metrics)))
-                for level in dataframe.index.levels[1]]
+        for level in dataframe.index.levels[1]:
+            metric_data = dict(self._recurse_dimensions(dataframe[:, level], dimensions[1:], metrics))
+
+            if not metric_data:
```

Comment:
If this check is necessary here, wouldn't it be necessary in the above loop on line 283?